ckks/evaluator.go

```diff
@@ -606,37 +563,37 @@ func (eval *evaluator) MultByConstAndAdd(ct0 *Ciphertext, constant interface{},
 
 		// If ctOut scaling is smaller than ct0's scale + the default scaling,
 		// then brings ctOut scale to ct0's scale.
-		if ctOut.scale < ct0.scale*scale {
+		if c1f64 < c0f64*scale {
 
-			if scale := math.Floor((scale * ct0.scale) / ctOut.scale); scale > 1 {
+			if scale := math.Floor((scale * c0f64) / c1f64); scale > 1 {
 
 				eval.MultByConst(ctOut, scale, ctOut)
 
 			}
 
-			ctOut.scale = scale * ct0.scale
+			ctOut.Scale().Mul(ct0.Scale(), scale, nil)
```

Comment:
todo: pass to remove use of interface method when receiver is a ciphertext type (Scale field is public).

Reply:
Done